crypto/src/lib.rs

```diff
@@ -12,16 +12,17 @@ mod signature;
 mod varint;
 
 #[cfg(not(feature = "std"))]
-use alloc::{format, string::String, vec::Vec};
+use alloc::{boxed::Box, format, string::String, vec::Vec};
 use core::{fmt, str::FromStr};
 
 #[cfg(feature = "base64")]
 pub use base64;
 use derive_more::{DebugCustom, Display};
 use getset::Getters;
 pub use hash::*;
-#[cfg(feature = "ffi")]
-use iroha_ffi::{ffi_export, IntoFfi, TryFromFfi};
+#[cfg(any(feature = "ffi_api", feature = "ffi"))]
```

Comment:
`ffi` and `ffi_api` are mutually exclusive since they do different transformations. `ffi_api` is actived in `iroha_core`, and `ffi` in `iroha_wasm`. When it becomes possible to remove the `std` feature I believe should also be able to remove the `ffi_api` feature

Reply:
mutual exclusiveness could be enforced in the `build.rs`

Reply:
Mutual exlusiveness should be documented. The names don't immediately tell me that I can have only one or the other. If they were called `ffi_std` and `ffi_no_std` it would be a little clearer (though the connection isn't strictly to `std`). We could call them 
- `ffi_std`/`ffi_no_std`
- `ffi_core`/`ffi_client`
- `ffi_core`/`ffi_contract`
- `ffi_no_wasm`/`ffi_wasm`

Reply:
As for the `ffi_api`, I was inspired by mutable_api feature name since it expands the API by exposing ffi functions.

Reply:
I introduced mutually exclusive features `ffi_import` and `ffi_export`. I hope the naming makes it more clear now

Reply:
Much clearer, thank you!